catalogs/catalog-jdbc-doris/src/main/java/com/datastrato/gravitino/catalog/doris/operation/DorisTableOperations.java

```diff
@@ -308,6 +311,22 @@ protected List<Index> getIndexes(Connection connection, String databaseName, Str
     }
   }
 
+  @Override
+  protected Transform[] getTablePartitioning(
+      Connection connection, String databaseName, String tableName) throws SQLException {
+    String showCreateTableSql = String.format("SHOW CREATE TABLE `%s`", tableName);
+    try (Statement statement = connection.createStatement();
+        ResultSet result = statement.executeQuery(showCreateTableSql)) {
+      StringBuilder createTableSql = new StringBuilder();
+      while (result.next()) {
+        createTableSql.append(result.getString("Create Table"));
+      }
+      Optional<Transform> transform =
+          DorisUtils.extractPartitionInfoFromSql(createTableSql.toString());
```

Comment:
It seems more efficient to pass only the line containing the `PARTITION BY` keyword to `DorisUtils.extractPartitionInfoFromSql` instead of concatenating and splitting SQL lines.

Reply:
We did not concatenate SQL here actually. When we execute `show create table` by JDBC, it will return the entire statement at once, which means the above loop will actually be executed once. So splitting SQL is necessary.
Besides, passing the entire statement to `DorisUtils.extractPartitionInfoFromSql` makes it easier for unit tests.

Reply:
If so, can we change above `WHILE` statement to `IF`?

Reply:
done